src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: string): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
+    const itemKey: string = item[key as keyof object];
```

Comment:
Дуже ненадійний опис типу
```suggestion
    const itemKey: string = item[key];
```